test/system/150-login.bats

```diff
@@ -398,6 +398,106 @@ EOF
     assert "$output" =~ "--retry-delay .*pull failures \(default \"5s\"\)"
 }
 
+@test "push and pull zstd chunked image" {
+    image1=localhost:${PODMAN_LOGIN_REGISTRY_PORT}/img1-$(safename)
+
+    globalopts="--pull-option enable_partial_pulls=true"
+    pushpullargs="--cert-dir ${PODMAN_LOGIN_WORKDIR}/trusted-registry-cert-dir \
+                  --creds ${PODMAN_LOGIN_USER}:${PODMAN_LOGIN_PASS}"
+
+    dockerfile1=$PODMAN_TMPDIR/Dockerfile.1
+    cat >$dockerfile1 <<EOF
+FROM $IMAGE
+RUN touch /
+EOF
+
+    run_podman $globalopts build --squash-all -t $image1 -f $dockerfile1 .
+    run_podman $globalopts push \
+               --compression-format zstd:chunked \
+               $pushpullargs \
+               $image1
+
+    run_podman $globalopts rmi $image1
+
+    run_podman $globalopts pull \
+               $pushpullargs \
+               $image1
+
+    run -0 skopeo inspect containers-storage:$image1
+    assert "$output" =~ "application/vnd.oci.image.layer.v1.tar\+zstd" "pulled image must be zstd-compressed"
+
+    dockerfile2=$PODMAN_TMPDIR/Dockerfile.2
+    cat >$dockerfile2 <<EOF
+FROM $image1
+RUN touch /new-file
+EOF
+
+    dockerfile3=$PODMAN_TMPDIR/Dockerfile.3
+    cat >$dockerfile3 <<EOF
+FROM $image1
+RUN touch /new-file-2
+EOF
+
+    image2=localhost:${PODMAN_LOGIN_REGISTRY_PORT}/img2-$(safename)
+    image3=localhost:${PODMAN_LOGIN_REGISTRY_PORT}/img3-$(safename)
+
+    run_podman $globalopts build -t $image2 -f $dockerfile2 .
+    run_podman $globalopts build -t $image3 -f $dockerfile3 .
+
+    run_podman $globalopts push \
+               --compression-format zstd:chunked \
+               $pushpullargs \
+               $image2
+
+    run_podman $globalopts push \
+               --compression-format zstd:chunked \
+               $pushpullargs \
+               $image3
+
+    run_podman $globalopts diff $image3 $image2
+    sorted_output=$(sort <<< $output | tr -d '\n')
+    assert "$sorted_output" = "A /new-file-2D /new-file"
+
+    run_podman $globalopts rmi $image2 $image3
+
+    run_podman $globalopts pull \
+               $pushpullargs \
+               $image2
+
+    run -0 skopeo inspect containers-storage:$image2
+    assert "$output" =~ "application/vnd.oci.image.layer.v1.tar\+zstd" "pulled image must be zstd-compressed"
+
+    run_podman $globalopts pull \
+               $pushpullargs \
+               $image3
+
+    run -0 skopeo inspect containers-storage:$image3
+    assert "$output" =~ "application/vnd.oci.image.layer.v1.tar\+zstd" "pulled image must be zstd-compressed"
+
+    run_podman $globalopts diff $image3 $image2
+    sorted_output=$(sort <<< $output | tr -d '\n')
+    assert "$sorted_output" = "A /new-file-2D /new-file"
+
+    for image in $image1 $image2 $image3; do
```

Comment:
Suggestion:
```diff
     for image in $image1 $image2 $image3; do
+        # Test for #24283: would fail with 'archive/tar: write too long'
+        run_podman save -o $PODMAN_TMPDIR/deleteme.tar $image
+        rm -f $PODMAN_TMPDIR/deleteme.tar
+
         push_dir=$PODMAN_TMPDIR/dir-image
```

Reply:
> ```diff
>      for image in $image1 $image2 $image3; do
> +        # Test for #24283: would fail with 'archive/tar: write too long'
> +        run_podman save -o $PODMAN_TMPDIR/deleteme.tar $image
> +        rm -f $PODMAN_TMPDIR/deleteme.tar
> +
>          push_dir=$PODMAN_TMPDIR/dir-image
> ```

thanks!  Since we are at it, I've added a podman save/load cycle to the test to stress more that partial pulled images must behave as any regular pulled image and we can move them around.